ethereum/attestation-test/contracts/VerifyTicket.sol

```diff
@@ -79,46 +84,34 @@ contract VerifyTicket {
     constructor()
     {
         owner = payable(msg.sender);
-        callCount = 0;
     }
 
     struct Length {
         uint decodeIndex;
         uint length;
     }
 
-    //Test function
-    function verifyTicketAttestationPayable(bytes memory attestation) public returns(bool)
+    function verifyTicketAttestation(bytes memory attestation, address attestor, address ticketIssuer) external view returns(address subject, bytes memory ticketId, bytes memory conferenceId, bool timeStampValid)
     {
-        address attestor;
-        (attestor,,,) = verifyTicketAttestation(attestation);
-        if (attestor != address(0))
-        {
-            callCount++;
-            return true;
-        }
-        else
-        {
-            revert();
-        }
+        ( subject, ticketId, conferenceId, timeStampValid ) = _verifyTicketAttestation( attestation, attestor, ticketIssuer);
     }
 
-    function verifyTicketAttestation(bytes memory attestation, address attestor, address ticketIssuer) public view returns(address subject, bytes memory ticketId)
+    function _verifyTicketAttestation(bytes memory attestation, address attestor, address ticketIssuer) public view returns(address subject, bytes memory ticketId, bytes memory conferenceId, bool timeStampValid)
     {
         //ensure the attestor and issuer keys are correct
         address recoveredAttestor;
-        address recoveredIsuer;
+        address recoveredIssuer;
 
-        (recoveredAttestor, recoveredIsuer, subject, ticketId) = verifyTicketAttestation(attestation);
+        (recoveredAttestor, recoveredIssuer, subject, ticketId, conferenceId, timeStampValid) = verifyTicketAttestation(attestation);
 
-        if (recoveredAttestor != attestor || recoveredIsuer != ticketIssuer)
+        if (recoveredAttestor != attestor || recoveredIssuer != ticketIssuer || !timeStampValid)
```

Comment:
Maybe this gets verified by the calling contract? But it is _crucial_ to validate that the `subject` returned here is the same as the key used to sign the transaction.

Reply:
> Maybe this gets verified by the calling contract? But it is _crucial_ to validate that the `subject` returned here is the same as the key used to sign the transaction.

Does the subject sign? In this attestation there's no signature from the subject (at least as far as I can see), only their public key. We have issuer and attestor signatures, which are checked.

Reply:
Okay, I see what you mean. But then we put a lot of responsibility on the external library to ensure that they use the address of `subject` to potentially transfer tokens to instead of the address that signed the request.

But I understand your point. Maybe we can and a comment in nice bold letters? :)

Reply:
> Okay, I see what you mean. But then we put a lot of responsibility on the external library to ensure that they use the address of `subject` to potentially transfer tokens to instead of the address that signed the request.
> 
> But I understand your point. Maybe we can and a comment in nice bold letters? :)

Ok I'll add that comment. 

Sometimes we might be calling the transaction from a server as a curtesy so the user doesn't need to spend gas (if they don't already have eth), in which case the transaction sender will be different from the destination of eg a minted token.

Reply:
> > Okay, I see what you mean. But then we put a lot of responsibility on the external library to ensure that they use the address of `subject` to potentially transfer tokens to instead of the address that signed the request.
> > But I understand your point. Maybe we can and a comment in nice bold letters? :)
> 
> Ok I'll add that comment.
> 
> Sometimes we might be calling the transaction from a server as a curtesy so the user doesn't need to spend gas (if they don't already have eth), in which case the transaction sender will be different from the destination of eg a minted token.

Yes, after your previous reply I realised that not validating the origin of the transaction matching the subject, could be considered a feature. So I think you are right, that it is best _not_ to check this in the smart contract, but instead document it.